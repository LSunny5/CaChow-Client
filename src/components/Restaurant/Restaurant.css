/*restaurant page*/
.restaurantContent {
    min-height: 92vh;
    padding: 10px;
    padding-bottom: 1rem;
}

/*box for restaurant information*/
.restaurantInfo {
    border: 2px solid black;
    padding: 10px;
    margin: 1rem 1rem 0;
    font-family: 'Delius', cursive;
    background: white;
    border-radius: 10px;
}

/*box for the restaurant menu*/
.restaurantMenu {
    border: 2px dotted black;
    padding: 10px;
    margin: 0 10px 10px;
    background: rgb(228, 248, 228);
    border-radius: 10px;
    font-family: 'Delius', cursive;
}

/*title for menu "menu"*/
.menuHeading {
    font-size: 1.8rem;
    font-family: 'Cinzel', serif;
}

/*container for all the menu items*/
.menuList {
    text-align: left;
    padding: 0;
    font-size: 1rem;
}

/*Name of restaurant*/
.restaurantName {
    font-size: 1.8rem;
    font-family: 'Cinzel', serif;
}

/*restaurant Type*/
.restaurantType {
    font-size: 0.6rem;
    color: darkgreen;
}

/*Hours title "hours"*/
.hoursHeading {
    font-family: 'Cinzel', serif;
}

.day {
    font-size: 0.8rem;
}

/*hours time text*/
.hoursDisplay {
    color: darkblue;
}

/*list style for menu*/
.bulletlist {
    list-style: none;
}

/*category row*/
.category {
    padding: 3px 10px;
    background: rgb(243, 193, 174);
}

/*item name*/
.foodName{
    list-style: none;
    position: relative;
    background: rgb(228, 248, 228);
}

/*price of item*/
.price {
    position: absolute;
    right: 0px;
}

/********************************************************************/
/*For Ipads and larger screens */
@media only screen and (min-width: 600px) {
    /*box for restaurant information*/
    .restaurantInfo {
        width: 50%;
        margin: 2rem auto 0;
    }

    /*box for the restaurant menu*/
    .restaurantMenu {
        width: 70%;
        margin: 1rem auto;
    }
}


/********************************************************************/
/*For Laptops and larger screens */
@media only screen and (min-width: 1024px) {
    /*box for restaurant information*/
    .restaurantInfo {
        width: 30%;
        margin: 1rem auto 0;
    }

    /*box for the restaurant menu*/
    .restaurantMenu {
        width: 50%;
        margin: 0 auto;
    }

    /*Name of restaurant*/
    .restaurantName {
        font-size: 2rem;
    }
}