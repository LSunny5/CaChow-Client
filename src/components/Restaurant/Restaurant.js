import React from 'react';
import './Restaurant.css';
import CachowContext from '../../CachowContext';
//import {findRestaurant, findHours} from '../../CachowHelpers';

class Restaurant extends React.Component {
    static contextType = CachowContext;

    findRestaurant = (restaurants = [], id) => {
        return restaurants.find(rest => rest.r_id === parseInt(id));
    }

    findHours = (hours = [], hoursId) => {
        return hours.find(h => h.hours_id === parseInt(hoursId));
    }

    getItems = (menu = [], restId) => {
        return (!restId) ? menu : menu.filter(item => item.item_restaurant === restId)
    };



    printItem = (item, categories, cat) => {
        for (let i = 1; i <= categories.length; i++) {
            if (parseInt(item.item_cat) === i) {

                    return (
                        <li key={item.item_id} className="category">
                            {categories[i - 1].cat_name}
                            <ul>
                                <li key={item.item_id} className="foodName">
                                    {item.item_name}
                                    <span className="price">{item.item_price}</span>
                                </li>

                            </ul>
                        </li>

                    )

                }
            }
        }
   


    render() {
        const { restaurants = [], hours = [], menu = [], categories = [] } = this.context;
        const { r_id } = this.props.match.params;

        const targetRestaurant = this.findRestaurant(restaurants, r_id) || {};
        const targetHours = this.findHours(hours, targetRestaurant.r_hours) || {};
        const targetItems = this.getItems(menu, r_id);





        //for every item in the category group > 0 print 

        return (
            <div className="restaurantContent">
                <section className="restaurantInfo">
                    <p className="restaurantName">({targetRestaurant.r_type})</p>
                    <h1 className="restaurantName">{targetRestaurant.r_name}</h1>
                    <p className="restaurantName">{targetRestaurant.r_address}</p>
                    <p className="restaurantName">{targetRestaurant.r_city}, {targetRestaurant.r_state}  {targetRestaurant.r_zip}</p>
                    <p className="restaurantName">{targetRestaurant.r_phone}</p>
                    <br />
                    <div className="hoursBox">
                        <h3 className="hoursHeading">Hours: </h3>
                        <p>Sunday: <span className="hoursDisplay">{targetHours.sun_open}</span> to <span className="hoursDisplay">{targetHours.sun_close}</span> </p>
                        <p>Monday: <span className="hoursDisplay">{targetHours.mon_open}</span> to <span className="hoursDisplay">{targetHours.mon_close}</span>  </p>
                        <p>Tuesday: <span className="hoursDisplay">{targetHours.tues_open}</span> to <span className="hoursDisplay">{targetHours.tues_close}</span> </p>
                        <p>Wednesday: <span className="hoursDisplay">{targetHours.wed_open}</span> to <span className="hoursDisplay">{targetHours.wed_close}</span> </p>
                        <p>Thursday: <span className="hoursDisplay">{targetHours.thu_open}</span> to <span className="hoursDisplay">{targetHours.thu_close}</span> </p>
                        <p>Friday: <span className="hoursDisplay">{targetHours.fri_open}</span> to <span className="hoursDisplay">{targetHours.fri_close}</span> </p>
                        <p>Saturday: <span className="hoursDisplay">{targetHours.sat_open}</span> to <span className="hoursDisplay">{targetHours.sat_close}</span> </p>
                    </div>
                </section>
                <br />
                <section className="restaurantMenu">
                    <h2 className="menuHeading">Menu</h2>
                    <ul className="menuList">

                        {targetItems.map(item => this.printItem(item, categories, item.item_cat))}





                        {/*  {targetItems.map(item =>
                        <li key={item.item_id} className="category">
                                {item.item_cat}
                                <ul>


                                    <li className="foodName">
                                        {item.item_name}
                                        <span className="price">${item.item_price}</span>


                                    </li>

                                </ul>
                            </li>
                        )
    } */}

                        {/* {for (let i = 1; i <= categories.length; i++) {
                                if (item.item_cat === i) {

                                    
                                        <li key={item.item_id} className="category">
                                            {item.item_cat}
                                            <ul>


                                                <li className="foodName">
                                                    {item.item_name}
                                                    <span className="price">${item.item_price}</span>


                                                </li>

                                            </ul>
                                        </li>


                                    




                                }  */}































                        {/*  {targetRestaurant.menu.forEach(item => {
                        <li> 
                            {menu.catId}
                        </li>
                    })
                     } */}

                        {/* <li className="category">
                            Burgers
                            <ul>
                                <li className="foodName">Single ShackBurger <span className="price">${targetRestaurant.menu[0].price}</span></li>
                                <li className="foodName">Single SmokeStack <span className="price">$7.39</span></li>
                            </ul>
                        </li>
                        <li className="category">Chicken
                            <ul>
                            <li className="foodName">Chick'n Shack <span className="price">$7.19</span></li>
                            </ul>
                        </li>
                        <li className="category">Sides
                            <ul>
                            <li className="foodName">Cheese Fries <span className="price">$4.09</span></li>
                            </ul>
                        </li>
                        <li className="category">Desserts
                            <ul>
                            <li className="foodName">S'mores Shake <span className="price">$5.99</span></li>
                            </ul>
                        </li>
                        <li className="category">Drinks
                            <ul>
                            <li className="foodName">Small Pink Lemonade <span className="price">$3.09</span></li>
                            </ul>
                        </li>
                     */}
                    </ul>
                </section>
            </div>
        );
    }
}

export default Restaurant;