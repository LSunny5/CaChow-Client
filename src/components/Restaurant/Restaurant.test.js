import React from 'react';
import ReactDOM from 'react-dom';
import Restaurant from './Restaurant';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`Restaurant component`, () => {
    const props = {
        r_id: 1,
        r_owner: 'OwnerS',
        r_image: 'NoneS',
        r_type: 'Fast',
        r_name: 'Restaurant Two',
        r_address: 'Address2',
        r_city: 'City2',
        r_state: 'State2',
        r_zip: 'Zip2',
        r_phone: 'Phone2',
        r_hours: 1,
        cat_id: 1,
        cat_name: 'Cat 1',
        hours_id: 1,
        sun_open: 'Open',
        sun_close: 'Closed',
        mon_open: 'Open',
        mon_close: 'Closed',
        tues_open: 'Open',
        tues_close: 'Closed',
        wed_open: 'Open',
        wed_close: 'Closed',
        thu_open: 'Open',
        thu_close: 'Closed',
        fri_open: 'Open',
        fri_close: 'Closed',
        sat_open: 'Open',
        sat_close: 'Closed',
        hours_owner: 'owner 1',
        item_id: 1,
        item_restaurant: 1,
        item_name: 'Item name',
        item_cat: 1,
        item_price: 1.23,
        match: {
            params: {
                r_id: 1
            }
        },
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Restaurant {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Restaurant form', () => {
        const wrapper = shallow(<Restaurant {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});