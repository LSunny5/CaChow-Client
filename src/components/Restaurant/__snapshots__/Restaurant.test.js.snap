// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant component renders the Restaurant form 1`] = `
<div
  className="restaurantContent"
>
  <section
    className="restaurantInfo"
  >
    <p
      className="restaurantName"
    >
      (
      )
    </p>
    <h1
      className="restaurantName"
    />
    <p
      className="restaurantName"
    />
    <p
      className="restaurantName"
    >
      , 
        
    </p>
    <p
      className="restaurantName"
    />
    <br />
    <div
      className="hoursBox"
    >
      <h3
        className="hoursHeading"
      >
        Hours: 
      </h3>
      <p>
        Sunday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
      <p>
        Monday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
          
      </p>
      <p>
        Tuesday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
      <p>
        Wednesday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
      <p>
        Thursday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
      <p>
        Friday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
      <p>
        Saturday: 
        <span
          className="hoursDisplay"
        />
         to 
        <span
          className="hoursDisplay"
        />
         
      </p>
    </div>
  </section>
  <br />
  <section
    className="restaurantMenu"
  >
    <h2
      className="menuHeading"
    >
      Menu Update
    </h2>
    <ul
      className="menuList"
    />
  </section>
</div>
`;
